juju/client/proxy/kubernetes/proxy.py

```diff
@@ -29,21 +30,18 @@ def __init__(
         self.namespace = namespace
         self.remote_port = remote_port
         self.service = service
+        self.temp_ca_file = None
+        self.port_forwarder = None
 
         try:
             self.remote_port = int(remote_port)
         except ValueError:
             raise ValueError(f"Invalid port number: {remote_port}")
 
-        self.port_forwarder = None
-
         if ca_cert:
-            self.temp_ca_file = tempfile.NamedTemporaryFile()  # noqa: SIM115
-            self.temp_ca_file.write(bytes(ca_cert, "utf-8"))
-            self.temp_ca_file.flush()
-            config.ssl_ca_cert = self.temp_ca_file.name
-        else:
-            self.temp_ca_file = None
+            with tempfile.NamedTemporaryFile(delete=False) as f:
+                f.write(bytes(ca_cert, "utf-8"))
+            self.temp_ca_file = config.ssl_ca_cert = f.name
```

Comment:
This would be clearer as two separate assignments:
```suggestion
            self.temp_ca_file = f.name
            config.ssl_ca_cert = f.name
```

Reply:
done.